src/private/WRLDBuildingApiHelpers.mm

```diff
@@ -0,0 +1,91 @@
+#include "WRLDBuildingApiHelpers.h"
+
+#import "WRLDBuildingHighlightOptions+Private.h"
+#import "WRLDBuildingDimensions+Private.h"
+#import "WRLDBuildingInformation+Private.h"
+#import "WRLDBuildingContour+Private.h"
+#import "WRLDMathApiHelpers.h"
+
+@interface WRLDBuildingApiHelpers ()
+
+@end
+
+@implementation WRLDBuildingApiHelpers
+{
+
+}
+
++ (Eegeo::BuildingHighlights::BuildingHighlightCreateParams)createBuildingHighlightCreateParams:(WRLDBuildingHighlightOptions*) buildingHighlightOptions
+{
+    CLLocationCoordinate2D selectionLocation = buildingHighlightOptions.selectionLocation;
+    Eegeo::Space::LatLong latLng = Eegeo::Space::LatLong::FromDegrees(selectionLocation.latitude, selectionLocation.longitude);
+
+    CGPoint selectionScreenPoint = buildingHighlightOptions.selectionScreenPoint;
+
+    int typeIntValue = static_cast<int>(buildingHighlightOptions.selectionMode);
+    Eegeo::BuildingHighlights::BuildingHighlightSelectionMode::Type type = static_cast<Eegeo::BuildingHighlights::BuildingHighlightSelectionMode::Type>(typeIntValue);
+
+    Eegeo::BuildingHighlights::BuildingHighlightCreateParams highlightCreateParams
+    {
+        type,
+        {latLng.GetLatitude(), latLng.GetLongitude()},
+        {static_cast<float>(selectionScreenPoint.x),static_cast<float>(selectionScreenPoint.y)},
+        [WRLDMathApiHelpers getEegeoColor:buildingHighlightOptions.color],
+        static_cast<bool>(buildingHighlightOptions.shouldCreateView)
+    };
+
+    return highlightCreateParams;
+}
+
++ (WRLDBuildingDimensions*) createWRLDBuildingDimensions:(const Eegeo::BuildingHighlights::BuildingDimensions&) withBuildingDimensions
+{
+    WRLDBuildingDimensions* buildingDimensions = [[WRLDBuildingDimensions alloc] initWithBaseAltitude:withBuildingDimensions.GetBaseAltitude()
+                                                                                          topAltitude:withBuildingDimensions.GetTopAltitude()
+                                                                                             centroid:CLLocationCoordinate2DMake(withBuildingDimensions.GetCentroid().GetLatitudeInDegrees(),
+                                                                                                                                 withBuildingDimensions.GetCentroid().GetLongitudeInDegrees())];
+    return buildingDimensions;
+}
+
++ (CLLocationCoordinate2D*)createWRLDBuildingContourPoints:(const std::vector<Eegeo::Space::LatLong>&) withPoints
+{
+    int pointCount = static_cast<int>(withPoints.size());
+    CLLocationCoordinate2D* points = new CLLocationCoordinate2D[pointCount];
```

Comment:
Is this ok, creating raw array via 'new', rather than populating some strongly-typed container and allocating via NSAlloc?
Elsewhere we model similar constructs with C++ std containers, eg:
https://github.com/wrld3d/ios-api/blob/master/src/private/WRLDPolygon.mm#L19
May be possible to take similar approach here with WRLDBuildingContour etc?